geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionManagerDUnitTest.java

```diff
@@ -159,77 +159,74 @@ public void testConnectAfterBeingShunned() {
-      long birthTime = startTime - timeout + gracePeriod;
-      MembershipManagerHelper.addSurpriseMember(sys, mbr, birthTime);
-      assertTrue("Member was not a surprise member", mgr.isSurpriseMember(mbr));
-
-      // force a real view change
-      SerializableRunnable connectDisconnect = new SerializableRunnable() {
-        public void run() {
-          getSystem().disconnect();
+            "<ExpectedException action=add>Removing shunned GemFire node</ExpectedException>");
+        try {
+          boolean accepted = mgr.addSurpriseMember(mbr);
+          Assert.assertTrue("member with old ID was not rejected (bug #44566)", !accepted);
+        } finally {
+          sys.getLogWriter().info(
+              "<ExpectedException action=remove>attempt to add old member</ExpectedException>");
+          sys.getLogWriter().info(
+              "<ExpectedException action=remove>Removing shunned GemFire node</ExpectedException>");
         }
-      };
-      vm0.invoke(connectDisconnect);
-
-      if (birthTime < (System.currentTimeMillis() - timeout)) {
-        return; // machine is too busy and we didn't get enough CPU to perform more assertions
-      }
-      assertTrue("Member was incorrectly removed from surprise member set",
-          mgr.isSurpriseMember(mbr));
+        mbr.setVmViewId(oldViewId);
+
+        // now forcibly add it as a surprise member and show that it is reaped
+        long gracePeriod = 5000;
+        long startTime = System.currentTimeMillis();
+        long timeout = ((GMSMembershipManager) mgr).getSurpriseMemberTimeout();
+        long birthTime = startTime - timeout + gracePeriod;
+        MembershipManagerHelper.addSurpriseMember(sys, mbr, birthTime);
+        assertTrue("Member was not a surprise member", mgr.isSurpriseMember(mbr));
+
+        if (birthTime < (System.currentTimeMillis() - timeout)) {
+          return; // machine is too busy and we didn't get enough CPU to perform more assertions
+        }
+        assertTrue("Member was incorrectly removed from surprise member set",
+            mgr.isSurpriseMember(mbr));
 
-      try {
-        Thread.sleep(gracePeriod);
-      } catch (InterruptedException e) {
-        fail("test was interrupted", e);
-      }
+        try {
+          Thread.sleep((timeout / 3) + gracePeriod);
+        } catch (InterruptedException e) {
+          fail("test was interrupted", e);
+        }
 
-      vm0.invoke(connectDisconnect);
-      assertTrue("Member was not removed from surprise member set", !mgr.isSurpriseMember(mbr));
+        assertTrue("Member was not removed from surprise member set", !mgr.isSurpriseMember(mbr));
 
-    } finally {
-      if (sys != null && sys.isConnected()) {
-        sys.disconnect();
+      } finally {
+        if (sys != null && sys.isConnected()) {
+          sys.disconnect();
+        }
       }
+    } finally {
+      System.getProperties().remove(DistributionConfig.GEMFIRE_PREFIX + "surprise-member-timeout");
```

Comment:
System.clearProperty(xxx) is better syntax, but I'd recommend deleting try-finally blocks like this and just add this Rule near the top of the dunit test:
```java
@Rule
public DistributedRestoreSystemProperties restoreSystemProperties = new DistributedRestoreSystemProperties();
```